pgdog/src/backend/pool/connection/binding.rs

```diff
@@ -237,6 +237,13 @@ impl Binding {
         Ok(())
     }
 
+    /// Execute a BEGIN on all servers
+    /// TODO: Block mutli-shard BEGINs as transaction should not occur on multiple shards
+    pub async fn begin(&mut self) -> Result<(), Error> {
+        let query = Query::new("BEGIN");
```

Comment:
See other comment. Client can pass options to `BEGIN`, changing transaction type.